File: lab2-functionalPrimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc a b = [a..b]

--Define applicatorFunc
applicatorFunc inpFunc a b s = if s=='s' then (sum inpFunc (a b)) else (sum inpFunc (a b))/((b - a) + 1) 

main = do
    let result = applicatorFunc inpFunc (2 25) 'a' --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show(result))

-- Brackets added to organise different operations i..e, adding the sum of (value)
-- Those same brackets are universal in terms of application, inpFunc parameters are bracketed
-- Main issues? inpFunc in main doesn't register as valid and "a" at the end of applicationFunc is invalid

File: lab2-functionalPrimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
let a;
let b;

function arrFunc(){
    let arr = [];
    for (let a = 1; a<=b; a++) {
        arr.push(a);
    }
    return arr;    
}

function applicatorFunc(inpFunc, a, b, s){
    if(s=='s'){
        const arr = inpFunc();        
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    }
    else{        
        const arr = inpFunc();
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum/b;
    }
}

let x = applicatorFunc(arrFunc, 1, 5, 's');
console.log(x);